src/app/lib/file_cache_handler.coffee

```diff
@@ -0,0 +1,217 @@
+async = require 'async'
+DesignDocuments = require '../replicator/design_documents'
+fs = require '../replicator/filesystem'
+log = require('./persistent_log')
+    prefix: "FileCacheHandler"
+    date: true
+
+
+instance = null
+
+
+module.exports = class FileCacheHandler
+
+
+    constructor: (@localDb, @replicateDb, @requestCozy) ->
+        return instance if instance
+        instance = @
+        @cache = {}
```

Comment:
It looks like you are using `@cache` as an hash. You should use `@cache = Object.create(null)` if it is the case. For example `@cache['toString']` will be truthy if instanciated as `{}`, but falsy (as expected) if instanciated as `Object.create(null)`.

See https://glebbahmutov.com/blog/sweet-naked-objects/ and http://www.devthought.com/2012/01/18/an-object-is-not-a-hash/